test_rmw_implementation/test/test_publisher.cpp

```diff
@@ -375,3 +375,75 @@ TEST_F(CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION), count_mismatched_subscri
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, subscription_count);
 }
+
+class CLASSNAME (TestPublisherUseLoan, RMW_IMPLEMENTATION)
+  : public CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION)
+{
+protected:
+  using Base = CLASSNAME(TestPublisherUse, RMW_IMPLEMENTATION);
+
+  void SetUp() override
+  {
+    Base::SetUp();
+    // Check if loaning is supported by the implementation
+    if (!pub->can_loan_messages) {
+      void * msg_pointer = nullptr;
+      rmw_publisher_allocation_t * null_allocation{nullptr};
+      rmw_ret_t ret = rmw_borrow_loaned_message(pub, ts, &msg_pointer);
+      EXPECT_EQ(RMW_RET_UNSUPPORTED, ret) << rmw_get_error_string().str;
+      rmw_reset_error();
+      EXPECT_EQ(nullptr, msg_pointer);
+      ret = rmw_return_loaned_message_from_publisher(pub, &msg_pointer);
+      EXPECT_EQ(RMW_RET_UNSUPPORTED, ret) << rmw_get_error_string().str;
+      rmw_reset_error();
+      EXPECT_EQ(nullptr, msg_pointer);
+      ret = rmw_publish_loaned_message(pub, &msg_pointer, null_allocation);
+      EXPECT_EQ(RMW_RET_UNSUPPORTED, ret) << rmw_get_error_string().str;
+      rmw_reset_error();
+      EXPECT_EQ(nullptr, msg_pointer);
```

Comment:
@Lobotuerk hmm, it's unusual to test in `SetUp` and then `GTEST_SKIP` the rest. What happens if one of these expectations fails?

Reply:
No idea, but I don't see another way of using both the variable and the functions returns

Reply:
Update: If any of the `EXPECTS` triggers an error, the test does not get skipped and it goes on to the tests on those implementations, which the ones not implemented give Errors. IMO seems a fair outcome.